lsat/store.go

```diff
@@ -0,0 +1,169 @@
+package lsat
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+var (
+	// ErrNoToken is the error returned when the store doesn't contain a
+	// token yet.
+	ErrNoToken = fmt.Errorf("no token in store")
+
+	storeFileName = "lsat.token"
+)
+
+// Store is an interface that allows users to store and retrieve an LSAT token.
+type Store interface {
+	// HasToken returns true if the store contains a token.
+	HasToken() bool
+
+	// CurrentToken returns the token that is currently contained in the
+	// store or an error if there is none.
+	CurrentToken() (*Token, error)
+
+	// AllTokens returns all tokens that the store has knowledge of, even
+	// if they might be expired.
+	AllTokens() ([]*Token, error)
+
+	// StoreToken saves a token to the store. Old tokens should be kept for
+	// accounting purposes but marked as invalid somehow.
+	StoreToken(*Token) error
+}
+
+// FileStore is an implementation of the Store interface that uses a single file
+// to save the serialized token.
+type FileStore struct {
+	fileName string
```

Comment:
Possible for multiple swaps to be in flight, so we should embed a `sync.Mutex` within the struct.